integration_tests/test_build_snap_grammar.py

```diff
@@ -0,0 +1,105 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import fileinput
+import subprocess
+
+import testscenarios
+from testtools.matchers import Contains, Equals
+
+import integration_tests
+import snapcraft
+
+
+def _construct_scenarios():
+    main_scenarios = {
+        'build-snap-grammar': True,
+        'build-snap-grammar-try': True,
+        'build-snap-grammar-try-skip': False,
+        'build-snap-grammar-try-else': True,
+        'build-snap-grammar-on': False,
+        'build-snap-grammar-on-else': True,
+    }
+
+    # Just some combinations
+    channel_scenarios = [
+        '', '/stable', '/latest/stable']
+
+    all_scenarios = []
+    for project, expected_install in main_scenarios.items():
+        for channel in channel_scenarios:
+            d = dict(project=project, hello_installed=expected_install,
+                     channel=channel)
+            scenario = ('{}{}'.format(project, channel), d)
+            all_scenarios.append(scenario)
+
+    return all_scenarios
+
+
+class BuildSnapGrammarTestCase(testscenarios.WithScenarios,
+                               integration_tests.TestCase):
+
+    scenarios = _construct_scenarios()
+
+    def setUp(self):
+        super().setUp()
+        if self._hello_is_installed():
+            self.fail(
+                'This integration test cannot run if you already have the '
+                "'hello' snap installed. Please uninstall it "
+                "by running 'sudo snap remove hello'.")
+
+    def tearDown(self):
+        super().tearDown()
+
+        # Remove hello. This is safe since the test fails if hello was already
+        # installed.
+        try:
+            subprocess.check_output(
+                ['sudo', 'snap', 'remove', 'hello'],
+                stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            self.fail("unable to remove 'hello': {}".format(e.output))
+
+    def _hello_is_installed(self):
```

Comment:
`_is_hello_installed` should be the name of this function.

Reply:
I replicated what was in `test_build_package_grammar`. Let's change this in a different PR.